compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -1239,7 +1239,8 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
         case res: Apply if res.symbol == defn.InternalQuoted_exprSplice
                         && level == 0
                         && !suppressInline =>
-          expandMacro(res.args.head, tree.span)
+          val expanded = expandMacro(res.args.head, tree.span)
+          typedExpr(expanded) // Inline calls and constant fold code generated by the macro
```

Comment:
I remember that retype checking of leaf nodes is problematic. I was surprised that the CI is green.

Reply:
I have been fixing those for a while aiming to fix this one.

Reply:
Did you have any specific one mind?

Reply:
I think name resolution will be a big problem, in particular in the presence of implicits.

For example, given the following program:

```Scala
def test(using c: Int) = {
  def foo(c: Int): Int = summon[Int]
}
```

After type check the body of `foo`, we get the tree `c`. Retype check the name `c` will change semantics of the program.

Reply:
The Inliner already retypes the tree without changing the initial elaboration. All implicit should already be resolved and not change in this tree.

Reply:
I see, because we are already in `ReTyper`.